src/main/java/Duke.java

```diff
@@ -1,10 +1,56 @@
+import java.io.File;
+import java.util.Scanner;
+
+/**
+ * Driver class which contains the main method of the program. Also encapsulates Duke,
+ * a to-do list. Each Duke object has a storage file, a list of tasks, and a user
+ * interface.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor.
+     * @param filePath file path of text file used to load and store tasks.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the application. A scanner is instantiated to read standard input.
+     */
+    public void run() {
+        ui.showWelcome();
+        Scanner sc = new Scanner(System.in);
+        while (sc.hasNextLine()) {
```

Comment:
i think doing it this way means that the "bye" command does not stop the app

Reply:
Actually "bye" does close the app. The app will exit with an exit code of 0. I invoked System.exit(0) in a method inside the Ui class. Just for your information, thank you for the comment though!